bmp388/bmp388.go

```diff
@@ -0,0 +1,246 @@
+package bmp388
+
+import (
+	"errors"
+	"math"
+
+	"tinygo.org/x/drivers"
+)
+
+type Oversampling byte
+type Mode byte
+type OutputDataRate byte
+type FilterCoefficient byte
+
+type BMP388Config struct {
+	Pressure         Oversampling
+	Temperature      Oversampling
+	Mode             Mode
+	ODR              OutputDataRate
+	IIR              FilterCoefficient
+	SeaLevelPressure int32 // sea level pressure in centipascals
+}
+
+type Device struct {
+	bus     drivers.I2C
+	Address uint8
+	cali    calibrationCoefficients
+	Config  BMP388Config
+}
+
+type calibrationCoefficients struct {
+	// Temperature compensation
+	t1 uint16
+	t2 uint16
+	t3 int8
+
+	// Pressure compensation
+	p1  int16
+	p2  int16
+	p3  int8
+	p4  int8
+	p5  uint16
+	p6  uint16
+	p7  int8
+	p8  int8
+	p9  int16
+	p10 int8
+	p11 int8
+
+	// temperature value used for pressure compensation
+	tlin int64
+}
+
+// New returns a bmp388 struct with the default I2C address. Configure must also be called after instanting
+func New(bus drivers.I2C) Device {
+	return Device{
+		bus:     bus,
+		Address: ADDRESS,
+	}
+}
+
+// Configure can enable settings on the BMP388 and reads the calibration coefficients
+func (d *Device) Configure(config BMP388Config) (err error) {
+	d.Config = config
+
+	if d.Config == (BMP388Config{}) {
+		d.Config.Mode = NORMAL
+	}
+
+	if d.Config.SeaLevelPressure == 0 {
+		d.Config.SeaLevelPressure = 10132500 // standard sea level pressure in cPa
+	}
+
+	// Turning on the pressure and temperature sensors and setting the measurement mode
+	err = d.writeRegister(REG_PWR_CTRL, PWR_PRESS|PWR_TEMP|byte(d.Config.Mode))
+
+	// Configure the oversampling, output data rate, and iir filter coefficient settings
+	err = d.writeRegister(REG_OSR, byte(d.Config.Pressure|d.Config.Temperature<<3))
+	err = d.writeRegister(REG_ODR, byte(d.Config.ODR))
+	err = d.writeRegister(REG_IIR, byte(d.Config.IIR<<1))
+
+	if err != nil {
+		return errors.New("bmp388: failed to configure sensor, check connection")
```

Comment:
Creating new error objects can be expensive. I believe they do not currently cause a heap allocation anymore, but they are still expensive. So can you please move these errors to global error values, such as this?

```go
var errConfiguration = errors.New("bmp388: failed to configure sensor, check connection")
```

Also note that there is no guarantee an error will be returned when the connection is broken.